File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/02_Extensibility/python/main.py
import sys
sys.path.append('..')
from vehicle import Vehicle, Car, Bike, Truck


# Extensible: Add new vehicle types without changing existing code
class Bus(Vehicle):
    def start(self):
        print("Starting the bus")


vehicles = [Car(), Bike(), Truck(), Bus()]
for vehicle in vehicles:
    vehicle.start()  # Polymorphic behavior handles the new type seamlessly

"""
Output:
Starting the car
Starting the bike
Starting the truck
Starting the bus
"""

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/02_Extensibility/javascript/main.js
const { Vehicle, Car, Bike, Truck } = require('../vehicle');

// Extensible: Add new vehicle types without changing existing code
class Bus extends Vehicle {
    start() {
        console.log("Starting the bus");
    }
}

const vehicles = [new Car(), new Bike(), new Truck(), new Bus()];
for (const vehicle of vehicles) {
    vehicle.start(); // Polymorphic behavior handles the new type seamlessly
}

/*
Output:
Starting the car
Starting the bike
Starting the truck
Starting the bus
*/

